zapcore/write_syncer.go

```diff
@@ -75,6 +77,53 @@ func (s *lockedWriteSyncer) Sync() error {
 	return err
 }
 
+type bufferWriterSyncer struct {
+	bufferWriter *bufio.Writer
+	ws           WriteSyncer
+}
+
+// bufferSize sizes the buffer associated with each log file. It's large
+// so that log records can accumulate without the logging thread blocking
+// on disk I/O. The flushDaemon will block instead.
```

Comment:
The logging thread can block on disk I/O if the buffer fills up, so I don't think this comment is accurate.

Repro:
https://play.golang.org/p/CB5qRcz84nW

If you change the limit from 1000 to 100, then there's no output (the prints are buffered), but eventually once the buffer is full, it calls the underlying `Write` method

Reply:
Yep, we set a large bufferSize to avoid blocking on disk I/O, however when flushing or buffer fills up, it will be blocked.